evm/src/executor/inspector/cheatcodes/ext.rs

```diff
@@ -160,21 +159,46 @@ fn get_env(key: &str, r#type: ParamType, delim: Option<&str>) -> Result<Bytes, B
         .map_err(|e| e.into())
 }
 
-fn read_file(path: &str) -> Result<Bytes, Bytes> {
-    let data = fs::read_to_string(path).map_err(|err| err.to_string().encode())?;
+macro_rules! check_path_allowed {
+    ($state:ident, $absolute_path:ident) => {
+        // Check if any of the allowed paths is the prefix for our absolute path.
+        if !$state
+            .config
+            .allowed_paths
+            .iter()
+            .any(|allowed_path| $absolute_path.starts_with(allowed_path))
+        {
+            return Err(util::encode_error("Path is not allowed."))
+        }
```

Comment:
I feel this should rather be function than a macro, how about 

`CheatsCondig::is_allowed(p) -> bool`

and `CheatsCondig::ensure_allowed(p) -> Result`

?

Reply:
agree